en/concept/architecture.md

```diff
@@ -23,50 +21,57 @@ section shows the components of the flight stack.
-> **Tip** For more information about each of these modules see the
+> **Tip** At runtime, you can inspect which modules are executed with the `top` command, 
+> and each module can be started/stopped individually via `<module_name> start/stop`.
+> For more information about each of these modules see the
 > [Modules & Commands Reference](../middleware/modules_main.md).
 
 The arrows show the information flow for the *most important* connections between
 the modules. In reality, there are many more connections than shown, and some data 
 (e.g. for parameters) is accessed by most of the modules.
 
-Modules communicate with each other through a publish-subscribe message bus
-named [uORB](../middleware/uorb.md). 
+Modules communicate with each other through the [uORB](../middleware/uorb.md)
+publish-subscribe message bus. 
 The use of the publish-subscribe scheme means that:
 
-- The system is reactive — it will update instantly when new data is available
+- The system is [reactive](http://www.reactivemanifesto.org) — it is
+  asynchronous and will update instantly when new data is available
 - All operations and communication are fully parallelized
 - A system component can consume data from anywhere in a thread-safe fashion
 
 > **Info** This architecture allows every single one of these
 > blocks to be rapidly and easily replaced, even at runtime.
 
 
-### Middleware {#middleware}
-
-The [middleware](../middleware/README.md) consists primarily of device drivers
-for embedded sensors, communication with the external world (companion computer,
-GCS, etc.) and the uORB publish-subscribe message bus.
-
-In addition, the middleware includes a [simulation layer](../simulation/README.md) 
-that allows PX4 flight code to run on a desktop operating system and control 
-a computer modeled vehicle in a simulated "world".
-
 ### Flight Stack {#flight-stack}
 
 The flight stack is a collection of guidance, navigation and control algorithms 
 for autonomous drones. 
 It includes controllers for fixed wing, multirotor and VTOL airframes 
 as well as estimators for attitude and position.
 
+The following diagram shows an overview of the building blocks of
+the flight stack. It contains the full pipeline from sensors, RC input and
+autonomous flight control (Navigator), down to the motor or servo control
+(Actuators).
+
+![PX4 High-Level Flight Stack](../../assets/diagrams/PX4_High-Level_Flight-Stack.svg)
+<!-- This diagram can be updated from 
+[here](https://drive.google.com/a/px4.io/file/d/15J0eCL77fHbItA249epT3i2iOx4VwJGI/view?usp=sharing) 
+and opened with draw.io Diagrams. You might need to request access if you
+don't have a px4.io Google account.
+Caution: it can happen that after exporting some of the arrows are wrong. In
+that case zoom into the graph until the arrows are correct, and then export
+again. -->
+
 An **estimator** takes one or more sensor inputs, combines them, and computes a
 vehicle state (for example the attitude from IMU sensor data).
 
```

Comment:
Do you think it is worth having a paragraph here on the navigator role?

Reply:
Hmm, not sure. The others are more generic terms while the navigator is a specific module.

Reply:
OK!